src/wayland/shell/mod.rs

```diff
@@ -14,5 +14,29 @@
 //! - The [`legacy`](legacy/index.html) module provides handlers for the `wl_shell` protocol, which
 //!   is now deprecated. You only need it if you want to support apps predating `xdg_shell`.
 
+use super::Serial;
+use thiserror::Error;
+
 pub mod legacy;
 pub mod xdg;
+
+/// Represents the possible errors returned from
+/// a surface operation
+#[derive(Debug, Error)]
+pub enum SurfaceError {
```

Comment:
Given this Error can only have a single variant, I'm not convinced of the need to make it an enum, rather than an unit struct. Furthermore, it may be worth making a generic error `DeadResource` alongside `utils::UnmanagedResource` and use that, rather than create a specific error for surfaces.

Reply:
I added a DeadResource error to utils and changed the shell to use it.